### ISSUE TITLE
Refresh DataView doesn't adjust height of subgroups

### ISSUE THREAD

**Issue by oinkspoink** [almende/vis#1441]
Apparently the implemented feature discussed in #339 has a bug with subgroups.

Example: https://jsfiddle.net/os00aaLz/

As you can see, the items are overlapping after refreshing the DataView, although they are in different subgroups. If you navigate horizontally afterwards, they are displayed correctly.

Is there a way to fix this without using something performance-heavy like timeline.redraw()?

Thanks

**Comment by AlexDM0** [comment]
While I agree this a bug, the fix behind the screens will call a redraw. We do not differentiate between the changes. @josdejong Could you shed some light on this?

Regards

**Comment by oinkspoink** [comment]
I tried to fix it by myself and figured out that the only thing I need from `timeline.redraw()`  is `this.itemSet.markDirty()` .

```javascript
  /**
   * Force a redraw. The size of all items will be recalculated.
   * Can be useful to manually redraw when option autoResize=false and the window
   * has been resized, or when the items CSS has been changed.
   *
   * Note: this function will be overridden on construction with a trottled version
   */
  Timeline.prototype.redraw = function () {
    this.itemSet && this.itemSet.markDirty({ refreshItems: true });
    this._redraw();
  };

```
So, I changed vis.js as follows:

```javascript
function ItemSet(body, options) {
...
    // listeners for the DataSet of the items
    this.itemListeners = {
      'add': function add(event, params, senderId) {
        me._onAdd(params.items);
        me.markDirty();
      },
      'update': function update(event, params, senderId) {
        me._onUpdate(params.items);
      },
      'remove': function remove(event, params, senderId) {
        me._onRemove(params.items);
        me.markDirty();
      }
    };
...
}
```

I'm sure there is a better solution and I'm not aware of some possible complications, but this worked for me.

**Comment by oinkspoink** [comment]
Meanwhile, I think my workaround is not that good, because its basically a second redraw. I'm curious what your solution would look like @josdejong.

**Comment by josdejong** [comment]
I've found the underlying cause: when a Group was being redrawn, first the items would be stacked again, and after that the height of groups and subgroups was updated. However when using subgroups, stacking of items depends on the height of the subgroup (which was updated *after* the stacking).

So I changed it to: first update the heights of the subgroups, then stack the items, then update the height of the group.
